### ISSUE TITLE
Adjust request poll intervall for API rate limiting

### ISSUE THREAD

**Issue by twiebe** [gridscale/gridscale_api_client_python#6]
Currently the synchronous API client will poll asynchronous requests every 0.5s. This makes hitting the API rate limit quite likely once you introduce concurrency.

For the API client to not feel sluggish, the poll intervall likely shouldn't simply be increased to a few seconds. It could be dynamically increased to a max and/or be request type-dependent.

**Comment by neur0manc** [comment]
Is there a way to implement web sockets in the back-end? This would make polling unnecessary. And feel much more native. The client would then just subscribe to the request_uuid and if something changes in the back-end the client is notified.

**Comment by bkircher** [comment]
> a way to implement web sockets in the back-end

Absolutely. Good point. Nothing concrete planned on that front though so polling it is until then.
